### PR TITLE
Syntax lookup for various decorators

### PR DESCRIPTION
**Pull request by kevanstannard**
Some suggestions for #162

I'm currently getting familiar with ReScript and related terminology so I'm not attached to any content in this PR, and I'd be grateful to receive guidance on improvements and corrections (including nitpicks).

Feel free to edit if you prefer.

Some of this content is derived from the old docs, otherwise it's a best effort based on my current understandings.

Thanks

### CODE REVIEW

File: misc_docs/syntax/decorator_deriving.mdx [hunk 1]
```diff
@@ -0,0 +1,202 @@
+---
+id: "deriving-decorator"
+keywords: ["deriving", "decorator", "record", "variant"]
+name: "@deriving"
+summary: "This is the `@deriving` decorator."
+category: "decorators"
+---
+
+The `@deriving(accessors)` decorator can be used to either:
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_deriving.mdx:9]: These are outdated examples and docs I believe..
Did you see https://rescript-lang.org/docs/manual/latest/generate-converters-accessors#convert-record-type-to-abstract-record?

The syntax discovery widget should be straight to the point, just point to related docs in a `References` section with a bullet list of links

File: misc_docs/syntax/decorator_as.mdx [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
-This is mostly useful to map to JS attribute names that cannot be expressed in ReScript (such as keywords).
+This is mostly useful to map to JavaScript attribute names that cannot be expressed in ReScript (such as keywords).
 
-### Example
+For example:
 
 <CodeTab labels={["ReScript", "JS Output"]}>
 
-```reason
+```re
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_as.mdx:17]:
```suggestion
```res
```

File: misc_docs/syntax/decorator_get.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+```js
+var div = document.createElement("div")
+var top = div.scrollTop
+```
+
+</CodeTab>
+
+
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_get.mdx:35]:
```suggestion
### References

- [Bind using Special `@bs` Getters & Setters](https://rescript-lang.org/docs/manual/latest/bind-to-js-object#bind-using-special-bs-getters--setters)
```

File: misc_docs/syntax/decorator_get_index.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+o["x"] = 1;
+o["y"] = 3;
+o["z"] = 5;
+
+var value = o["y"];
+```
+
+</CodeTab>
+
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_get_index.mdx:72]:
```suggestion
### References
- [Bind using Special `@bs` Getters & Setters](https://rescript-lang.org/docs/manual/latest/bind-to-js-object#bind-using-special-bs-getters--setters)
```

File: misc_docs/syntax/decorator_inline.mdx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+}
+```
+
+```js
+if (process.env.NODE_ENV === "development") {
+  console.log("development");
+}
+```
+
+</CodeTab>
```
Comment (ryyppy) [inline misc_docs/syntax/decorator_inline.mdx:93]:
```suggestion
</CodeTab>

### References

- [Attribute (Decorator)](https://rescript-lang.org/docs/manual/latest/attribute)
```

**Comment by ryyppy** [comment]
not gonna lie, this is really really good.  Some topics might need some fine tuning, but this is just nitpicking. Thanks for your hard work!

I am considering merging this and maybe even merge the current state of the syntax widget branch into master already, since decorators are currently one of the harder topics, and it might be good for our users to have access to a nice index.

let me know when you are ready with this, and i will take some time doing a double check before putting it in production. Future PRs will probably just go into master later on (just in case if you want to work on more stuff).

**Comment by kevanstannard** [comment]
@ryyppy thanks for your feedback, was very helpful, particularly clarifying the goals of the syntax lookup vs core documentation. I've cleaned up all of these pages and tried to standardise; simple description, one example, plus links to references. Is that closer to what you had in mind?

Any further feedback is welcome, and feel free to modify as you like.

Also, I wasn't sure how best to indicate the status of `@deriving`, but take a look and see what you think.

There are a couple of remaining decorators not included in this PR that I wasn't sure about, so I'll leave them out for now.

Otherwise, ready for another review 👍

**Comment by ryyppy** [comment]
yeah way cleaner and uniform now. The references also really make the difference!

Will merge this later today, then do some further merges into master, and will post an update in our core issue and tick all the decorators that were added here. We will tackle the rest later on if necessary!

**Pull request merged**